pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/Patch.java

```diff
@@ -142,12 +142,12 @@ public static boolean patch(@NonNull String id, @NonNull String user, @NonNull N
                 if (result == 1) {
                     logger.info("Updated " + dbElement.getTableName() + " record w/ id " + id);
                 } else {
-                    throw new RuntimeException(
-                            "Error updating " + dbElement.getTableName() + " record of w/ id " + id + " it was updating " + result
-                                    + " rows");
+                    throw new DsmInternalError(toErrorMessage(dbElement.getTableName(), dbElement.getColumnName(),
+                            nameValue.getValue(), dbElement.getPrimaryKey(), id) + ": " + result + " rows updated");
                 }
             } catch (SQLIntegrityConstraintViolationException ex) {
-                throw new DuplicateException(dbElement.getColumnName());
+                throw new DsmInternalError(toErrorMessage(dbElement.getTableName(), dbElement.getColumnName(),
+                        nameValue.getValue(), dbElement.getPrimaryKey(), id), ex);
             } catch (SQLException ex) {
                 dbVals.resultException = ex;
             }
```

Comment:
Nit: Does it make sense to convert the RuntimeException below to DsmInternalError?

Reply:
Done!